src/utils.ts

```diff
@@ -51,11 +54,55 @@ export function isBooleanValue(value: string): boolean {
  * @param {string} value - The string to be converted.
  * @return {boolean|null} - The converted boolean value if the string is recognized as representing a boolean, otherwise null.
  */
-export function stringToBoolean(value: string) {
+export function stringToBoolean(value: string): boolean | null {
   const sanitized = value.toLowerCase().trim();
   const thrutyValues = ["true", "t", "1"];
   const falsyValues = ["false", "f", "0"];
   if (thrutyValues.includes(sanitized)) return true;
   else if (falsyValues.includes(sanitized)) return false;
   else return null;
 }
+
+/**
+ * Finds an association within an array of IncludeObject recursively.
+ * @param includes - Array of IncludeObject representing associations
+ * @param association - Association name to search for within the IncludeObject array
+ * @returns {IncludeObject | null} The found IncludeObject for the specified association, or null if not found
+ */
+export function findInclude(includes: IncludeObject[], association: string): IncludeObject | null {
+  for (let i = 0; i < includes.length; i++) {
+    const item = includes[i];
+    if (item.association === association) {
+      return item;
+    }
+    if (item.include) {
+      const innerInclude = findInclude(item.include, association);
+      if (innerInclude) return innerInclude;
+    }
+  }
+  return null;
+}
+
+/**
+ * Retrieves attribute names from a Sequelize model's raw attributes whose types
+ * are included in the types array.
+ * @param model - Sequelize Model class
+ * @param {string[]} types - An array of Sequelize attribute types, ex.: ["STRING", "TEXT"].
+ * @returns {string[]} Array of attribute names with string or text data type
+ */
+export function getAttributesByTypes(
+  model: typeof Model,
+  types: string[]
+): string[] {
+  // Retrieve all attribute names from the rawAttributes of the Sequelize model
```

Comment:
Remove inline comments

Reply:
Why?

Reply:
If the function's jsdoc is clear and precise on what the function does, commenting each line of code is unnecesary. In most cases overcomenting makes the code unclear and is a bad practice. The code in this function speaks for itself, it's clear on what it does.